lib/default_hostgroup_base_host_patch.rb

```diff
@@ -46,9 +46,17 @@ def group_matches?(facts)
       host_fact_value = facts_hash[fact_name]
       Rails.logger.info "Fact = #{fact_name}"
       Rails.logger.info "Regex = #{fact_regex}"
-      return true if Regexp.new(fact_regex).match(host_fact_value)
+      if Setting[:use_and_operator] == true
+        return false if not Regexp.new(fact_regex).match(host_fact_value)
+      else
+        return true if Regexp.new(fact_regex).match(host_fact_value)
+      end
+    end
+    if Setting[:use_and_operator] == true
+      true
+    else
+      false
```

Comment:
Since you're saying `if true, return true`, then this can be simplified to just
```
return Setting[:use_and_operator]
```
That avoids the if block entirely.